Classes/Cache/Frontend/VariableFrontendWithWarmupReservation.php

```diff
@@ -63,20 +62,20 @@ public function flush()
     public function flushByTag($tag)
     {
         parent::flushByTag($tag);
-        $warmupReservationService = GeneralUtility::makeInstance(WarmupReservationService::class);
-        $urls = $warmupReservationService->collectReservationsByCacheTags($this->getIdentifier(), [$tag]);
-        $queueService = GeneralUtility::makeInstance(QueueService::class);
-        $queueService->queueMany($urls);
+
+        $queueMakerService = GeneralUtility::makeInstance(QueueMakerService::class);
+        $queueMakerService->addToQueue($this->getIdentifier(), 1, $tag);
     }
 
     public function flushByTags(array $tags)
     {
         parent::flushByTags($tags);
-        $tags = array_unique($tags);
-        $warmupReservationService = GeneralUtility::makeInstance(WarmupReservationService::class);
-        $urls = $warmupReservationService->collectReservationsByCacheTags($this->getIdentifier(), $tags);
-        $queueService = GeneralUtility::makeInstance(QueueService::class);
-        $queueService->queueMany($urls);
+
+        $queueMakerService = GeneralUtility::makeInstance(QueueMakerService::class);
+
+        foreach ($tags as $tag) {
+            $queueMakerService->addToQueue($this->getIdentifier(), 2, $tag);
```

Comment:
what do the types `0`, `1` and `2` mean? can you introduce speaking class constants for them (something like `QueueMakerService::CACHE_ENTRY_TYPE_ALL = 0`)?

Reply:
Class constants added.